### PR TITLE
deploy: move vol-rep crds to kubernetes-csi-addons

### PR DESCRIPTION
**Pull request by yati1998**
This PR moves the volume replication operator to kubernetes-csi-addons.
This includes copying the types.go file and use it to generate the crds
and add rbac rules

Signed-off-by: yatipadia <ypadia@redhat.com>

### CODE REVIEW

**Review by Rakshith-R — CHANGES_REQUESTED** [review]
follow the steps to transition current apis and controller to multigroup folder layout
https://kubebuilder.io/migration/multi-group.html

**Comment by nixpanic** [comment]
It seems a few files were modified/created, and those need to be checked in as well.

**Comment by yati1998** [comment]
> follow the steps to transition current apis and controller to multigroup folder layout https://kubebuilder.io/migration/multi-group.html

Hey, @Rakshith-R do you mean for adding controllers/ part , didn't get what is missed.

File: apis/replication.storage/v1alpha1/volumereplication_types.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2021.
+Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (Rakshith-R) [inline apis/replication.storage/v1alpha1/volumereplication_types.go:3]: This needs to be changed

File: apis/replication.storage/v1alpha1/volumereplicationclass_types.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2021.
+Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (Rakshith-R) [inline apis/replication.storage/v1alpha1/volumereplicationclass_types.go:3]: same here

**Review by Rakshith-R — CHANGES_REQUESTED** [review]

**Comment by yati1998** [comment]
@Rakshith-R @nixpanic @Madhu-1

**Review by Rakshith-R — CHANGES_REQUESTED** [review]
Please add docs similar to reclaimspace and network fence
https://github.com/csi-addons/kubernetes-csi-addons/tree/main/docs

**Review by Madhu-1 — DISMISSED** [review]
changes LGTM.

**Review by Rakshith-R — APPROVED** [review]

**Review by Madhu-1 — APPROVED** [review]

**Pull request merged**